packages/common/src/utils.ts

```diff
@@ -116,6 +116,7 @@ export const decodeIncomingBody = (body: string): string | undefined => {
       utils.RLP.decode(new Uint8Array(JSON.parse(`[${body}]`)))[0]
     );
   } catch {
-    throw new Error("failed to decode body");
```

Comment:
make this throw, and catch it wherever `decodeIncomingBody` is used in the codebase